src/main/scala/view/scalaFX/components/WolfView.scala

```diff
@@ -0,0 +1,115 @@
+package view.scalaFX.components
+
+import engine.SimulationConstants.WOLVES_PHASE
+import scalafx.Includes.{at, double2DurationHelper}
+import scalafx.animation.{KeyFrame, Timeline}
+import scalafx.scene.image.{Image, ImageView}
+import scalafx.util.Duration
+import view.scalaFX.ScalaFxViewConstants.{PANEL_SKY_ZONE, PREFERRED_BUNNY_PANEL_HEIGHT, PREFERRED_BUNNY_PANEL_WIDTH, PREFERRED_BUNNY_SIZE}
+import view.scalaFX.utilities.Direction
+import view.scalaFX.utilities.Direction._
+
+import scala.language.postfixOps
+import scala.util.Random
+
+/** Bunny wrapper in order to manage its movement inside of the GUI */
+trait DisturbingFactorView {
+  /** Reference to the model disturbing factor entity */
+//  val wolf: DisturbingFactor
+
+  /** The image of the disturbing factor displayed on the GUI */
+  val imageView: ImageView
+
+  /** The Direction of the disturbing factor */
+  var direction: Direction
+
+  /** The X-Axis position of the disturbing factor */
+  var positionX: Double
+
+  /** The Y-Axis position of the disturbing factor */
+  var positionY: Double
+
+  /** Type annotation for a Seq of KeyFrames */
+  type AnimationFrames = Seq[KeyFrame]
+
+  /** Starts the disturbing factor animation */
+  def play(): Unit
+}
+
+object WolfView {
+
+  def apply(): DisturbingFactorView = {
+    val newX = Random.nextInt(PREFERRED_BUNNY_PANEL_WIDTH)
+    val newY = Random.nextInt(PREFERRED_BUNNY_PANEL_HEIGHT) + PANEL_SKY_ZONE
+
+    WolfViewImpl(new ImageView {
+      image = new Image("/img/factors/wolf.png")
+      x = newX
+      y = newY
+      fitWidth = PREFERRED_BUNNY_SIZE
+      fitHeight = PREFERRED_BUNNY_SIZE
+      preserveRatio = true
+      scaleX = Direction.scaleXValue(Right)
+    }, Right, newX, newY)
+  }
+
+  private case class WolfViewImpl( imageView: ImageView,
+                                   var direction: Direction,
+                                   var positionX: Double,
+                                   var positionY: Double) extends DisturbingFactorView {
+
+    private val wolfImage: Image = new Image("/img/factors/wolf.png")
+    private var movingSpace = 50
+
+    private val timeline: Timeline = new Timeline {
+      onFinished = _ => {
+        keyFrames = move()
+        this.play()
+      }
+      delay = Duration(WOLVES_PHASE)
+      autoReverse = true
+      cycleCount = 1
+      keyFrames = move()
+    }
+
+    private def move(): AnimationFrames = {
+      checkDirection()
+      Seq(
+        at(0 s){
+          Set(imageView.image -> wolfImage)
+        },
+        at(0.5 s) {
+          moveHorizontally()
+          Set(imageView.x -> positionX, imageView.y -> positionY)
+        },
+        at(1 s) {
+          moveHorizontally()
+          Set(imageView.x -> positionX, imageView.y -> positionY)
+        },
+        at(1.1 s) {
+          Set(imageView.image -> wolfImage)
+        }
+      )
+    }
```

Comment:
Questa timeline è concettualmente sbagliata per il movimento dei lupi, perchè copiata troppo dalla logica del salto del coniglio.

Il salto del coniglio si suddivide in 3 parti: il cambiamento dell'immagine in quella che salta, la prima metà del salto e la seconda metà. Una volta svolto il movimento l'immagine viene rimessa quella del conglio fermo. A te tutta questa logica non serve perchè il lupo deve solo muoversi in orizzontale.